### ISSUE TITLE
Resizing a window doesn't always update the layout

### ISSUE THREAD

**Issue by sdutoit** [koekeishiya/yabai#1199]
Sometimes, when I resize a window, the layout doesn't seem to update and other windows in the space remain at the same size.

This has been happening for me for as long as I can remember using yabai. I just reproduced it with the 4.0.0 release.

E.g. in this case, I have two windows tiled using yabai, and after resizing the window on the left (vscode) to be a little smaller, the window on the right (iterm2) remains as it is:

![yabai-resize-fail](https://user-images.githubusercontent.com/819177/158923601-3ba333c9-fe6e-44e2-b34a-e8a5eb427d53.png)

This is on macOS Catalina 10.15.7 (19H1715), though it's happened for me with earlier macOS releases as well. SIP is NOT disabled.

**Comment by sdutoit** [comment]
PS: Thanks for building yabai, it's amazing!

**Comment by koekeishiya** [comment]
This is a bit tricky to solve, because the issue actually exists because of the order the events are received in.

In the scenario in which the bug happens, we get the following events:

```
EVENT_HANDLER_MOUSE_DOWN: 688.58, 460.39
EVENT_HANDLER_MOUSE_DRAGGED: 689.14, 460.39
EVENT_HANDLER_MOUSE_DRAGGED: 694.32, 460.39
EVENT_HANDLER_MOUSE_DRAGGED: 724.27, 459.79
EVENT_HANDLER_MOUSE_UP: 724.27, 459.79
EVENT_HANDLER_WINDOW_RESIZED: kitty 5746
```

The issue is that the mouse_up event is received before the window_resized event. We use the mouse_up event to determine that the  user has finished making adjustments and it is the end-marker triggering the layout update. 

In this case however, we have not yet received information that the window size has changed, so we do not know that the layout needs to be updated. If that makes sense?

Not sure yet if this is fixable.

**Comment by sdutoit** [comment]
Thanks for the explanation!

Maybe I'm missing something here, but can't we just treat a window resize event happening outside of a mousedown->mouseup section as a layout updater trigger? I imagine we only get one resize event after the mouse up event?

**Comment by koekeishiya** [comment]
I removed the cached value and checked against the live frame returned by the AX API and it did not resolve the issue. There appears to be some issue where the AX API requests that we perform are not always respected; they are simply ignored by the target window. This is also what's happening when a dragged window is not snapped back to the assigned position - we do call the AX API, but the request is ignored.

**Comment by koekeishiya** [comment]
Fixed in v7.0.0.
